### DISCUSSION TITLE
how to clean data of prometheus PVC

### DISCUSSION THREAD

**Discussion by jainpratik163** [prometheus-operator/prometheus-operator#4133] [Q&A] [answered]
we have  install Grafana and Prometheus using the helm chart in the AWS K8s cluster and also we have configured PVC of 50Gb but soon this PVC got full and then we tried to delete the pod but still we face space issue.
then we increase the PVC size , is there way where we can auto clean the PVC.
and also when we delete the pod PVC data should also clean.

can anyone please suggest how we can handle such thing?

**Comment by paulfantom** [comment, marked as answer]
You can configure `retention` or `retentionSize` in prometheus to automatically delete data. If you want to remove all data on demand, remove PV and allow storage controller to reprovision it.

**Reply by Stewary-sudo** [reply]
Where we would find the retention period as we are using Kube-prometheus-stack

**Reply by Andrewq11** [reply]
Within the prometheusSpec object in the kube-prometheus-stack chart's values.yaml file. Look for the `retention` key.
